src/value.h

```diff
@@ -3,20 +3,32 @@
 #include <vector>
 #include <string>
 #include <algorithm>
+#include <cstdint>
 #include <limits>
 
 // Workaround for https://bugreports.qt-project.org/browse/QTBUG-22829
 #ifndef Q_MOC_RUN
 #include <boost/variant.hpp>
 #include <boost/lexical_cast.hpp>
 #include <glib.h>
-
 #endif
-#include <cstdint>
+
+#include "AST.h"
 #include "memory.h"
 
 class tostring_visitor;
 class tostream_visitor;
+class ValuePtr;
+class Context;
+
+class Expression : public ASTNode
```

Comment:
There is no need to move the `Expression` class from its own header here - you can use a forward declaration, as all references to it are using a `shared_ptr`

Reply:
Ahh, good point. I moved it before the `shared_ptr` change. Moved back.